tests/FromCommunity/IssueWithQueryArrayTest.php

```diff
@@ -12,62 +12,183 @@ final class IssueWithQueryArrayTest extends TestCase
     }
 
-    protected function makeYaml(string $style, string $type, string $format) : string
+    public function testConvertSingleLayerDeepObject() : void
+    {
+        $yaml      = /** @lang yaml */
+            <<<YAML
+openapi: 3.0.0
+info:
+  title: Product import API
+  version: '1.0'
+servers:
+  - url: 'http://localhost:8000/api/v1'
+paths:
+  /users:
+    get:
+      parameters:
+        - in: query
+          name: id
+          required: true
+          style: deepObject
+          explode: true
+          schema:
+            type: object
+            properties:
+              before:
+                type: integer
+                format: int32
+              after:
+                type: integer
+                format: int32
+      responses:
+        '200':
+          description: A list of users
+YAML;
+        $validator = (new ValidatorBuilder())->fromYaml($yaml)->getServerRequestValidator();
+        $validator->validate($this->makeRequest('deepObject', 'integer'));
+        $this->addToAssertionCount(1);
+    }
+
+    public function testConvertMultiLayerDeepObject() : void
+    {
+        $yaml      = /** @lang yaml */
+            <<<YAML
+openapi: 3.0.0
+info:
+  title: Product import API
+  version: '1.0'
+servers:
+  - url: 'http://localhost:8000/api/v1'
+paths:
+  /users:
+    get:
+      parameters:
+        - in: query
+          name: id
+          required: true
+          style: deepObject
+          explode: true
+          schema:
+            type: object
+            properties:
+              before:
+                type: object
+                properties:
+                  first:
+                    type: object
+                    properties:
+                      second:
+                        type: integer
+                        format: int32
+              after:
+                type: integer
+                format: int32
+      responses:
+        '200':
+          description: A list of users
+YAML;
+        $validator = (new ValidatorBuilder())->fromYaml($yaml)->getServerRequestValidator();
+        $validator->validate($this->makeRequest('deepObject', 'deep'));
+        $this->addToAssertionCount(1);
+    }
+
+    public function testConvertDeepObjectError() : void
+    {
+        $yaml = /** @lang yaml */
+            <<<YAML
+openapi: 3.0.0
+info:
+  title: Product import API
+  version: '1.0'
+servers:
+  - url: 'http://localhost:8000/api/v1'
+paths:
+  /users:
+    get:
+      parameters:
+        - in: query
+          name: id
+          required: true
+          style: deepObject
+          explode: true
+          schema:
+            type: object
+            properties:
+              before:
+                type: integer
+                format: int32
+              after:
+                type: integer
+                format: int32
+      responses:
+        '200':
+          description: A list of users
+YAML;
+        $this->expectExceptionMessage('Value "{
```

Comment:
I'm not sure whether it's good explanation of what is going wrong with parameter. Can we also inspect prev exceptions to examine correct validations ?

Reply:
Updated error tests to inspect previous exceptions instead of error message.